docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+
+Now we consider the consensus layer as an option for dealing with this, but unfortunately, **there is no way to resolve it in the consensus layer, ideally**. Unlike Byzantine or 'dishonest' behavior which can be directly verified by some cryptographic evidence (e.g. double pre-votes), 'unfaithful' or 'bad' leader can not be handled by the consensus because it's not a violation of the consensus protocol.
+
+### Leader-based vs Leaderless
+
+One might think that a 'leaderless' consensus might be the solution for this. Basically, it might be, but there is a critical problem: network complexity.
+In a leader-based consensus, the network can (in the best case) progress if the validators turn their nodes exactly two times (i.e., two times of broadcasts). However, in a leaderless consensus, the number of broadcasts is proportional to the number of validators. This will never be acceptable in our essential assumption of light-weight node operation.
+
+TODO: check it and add reference
+
+### Veto the Round
+
+Again it can't be handled by the consensus layer because neither laziness nor censorship is a violation of the consensus protocol. However, we can find a detour if we **exploit the network asynchrony**. If a validator concludes that the current round should be escaped because of the honest-but-lazy-or-bad leader, they can just **pretend that the round is timed out**. This provides a very special feature over plain Tendermint; **veto the leader**.
+
+Since Tendermint is based on rounds, there is a special mechanism called *nil-vote*. Nodes can cast a nil-vote after the timeout, to escape the current round. This works similarly to the non-nil case, in that it takes two steps.
+
+In plain Tendermint, if the validators use nil-votes to veto the proposer, it just works because it's not so different from a rash timeout, but it might get stuck until the timeout in the split case. (Think about the case of 6/10 non-nil-votes and 3/10 nil-votes, 1/10 Unresponsive Byzantine nodes)
+Again, because we're using very long rounds, it will waste a lot of time, waiting for the timeout of the round in that kind of situation. If we encounter this unlucky case of a split, it is no better than the governance's solution (no approval of agendas).
+
+### Vetomint
+
+To improve this, we propose a variation of Tendermint, called *Vetomint*, which is effective in the 'split' case.
+
+### Properties
+
+1. `<1/6` BFT
+2. It is allowed (i.e., considered as a non-byzantine behavior) to cast a nil-vote before the timeout, with *some kind of reason*.
+3. If all the honest nodes cast either a vote or nil-vote, it is guaranteed that the round progresses; the **early termination**.
```

Comment:
In this sentence and below, how about consistently using "non-nil-vote" for some clarity?

Reply:
updated